From 1d27f30e0ee3614f1e5d64e17c65d58b9de99990 Mon Sep 17 00:00:00 2001
Message-Id: <1d27f30e0ee3614f1e5d64e17c65d58b9de99990.1684832810.git.stefan@agner.ch>
In-Reply-To: <c46f6cb84fb189da723a9932a5195a553762e2b6.1684832810.git.stefan@agner.ch>
References: <c46f6cb84fb189da723a9932a5195a553762e2b6.1684832810.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Tue, 23 May 2023 10:50:24 +0200
Subject: [PATCH] [python] make enum part of cluster wheel

The new enum type is required in the Cluster objects. Make it and its
dependency part of the Cluster CHIP wheel.

Fixes PR #24352.
---
 src/controller/python/BUILD.gn | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/controller/python/BUILD.gn b/src/controller/python/BUILD.gn
index d9cb4e588f..bcbdaa4203 100644
--- a/src/controller/python/BUILD.gn
+++ b/src/controller/python/BUILD.gn
@@ -223,7 +223,6 @@ chip_python_wheel_action("chip-core") {
         "chip/clusters/Attribute.py",
         "chip/clusters/Command.py",
         "chip/clusters/__init__.py",
-        "chip/clusters/enum.py",
         "chip/commissioning/__init__.py",
         "chip/commissioning/commissioning_flow_blocks.py",
         "chip/commissioning/pase.py",
@@ -309,7 +308,6 @@ chip_python_wheel_action("chip-core") {
   }
 
   py_package_reqs = [
-    "aenum",
     "coloredlogs",
     "construct",
     "dacite",
@@ -358,6 +356,7 @@ chip_python_wheel_action("chip-clusters") {
         "chip/ChipUtility.py",
         "chip/clusters/CHIPClusters.py",
         "chip/clusters/ClusterObjects.py",
+        "chip/clusters/enum.py",
         "chip/clusters/Objects.py",
         "chip/clusters/TestObjects.py",
         "chip/clusters/Types.py",
@@ -378,7 +377,10 @@ chip_python_wheel_action("chip-clusters") {
     "chip.tlv",
   ]
 
-  py_package_reqs = [ "dacite" ]
+  py_package_reqs = [
+    "dacite",
+    "aenum",
+  ]
 
   py_package_name = "${chip_python_package_prefix}-clusters"
   py_package_output = string_replace(py_package_name, "-", "_")
-- 
2.40.1

